viewing_party/party.py

```diff
@@ -1,23 +1,174 @@
+        "genre": genre,
+        "rating": rating
+    }
+    return movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    sum = 0.0
+    len_arr = len(user_data["watched"])
+
+    if len_arr == 0:
+        return sum
+
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+
+    return sum/len_arr
+
+
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) == 0:
+        return None
+
+    genre_count = {}
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        genre_count[genre] = genre_count.get(genre, 0) + 1
+
+    max_count = max(genre_count.values())
+
+    for k, v in genre_count.items():
+        if v == max_count:
+            return k
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+# USER_DATA_3["friends"] =  [
+#         {
+#             "watched": [
+#                 FANTASY_1,
+#                 FANTASY_3,
+#                 FANTASY_4,
+#                 HORROR_1,
+#             ]
+#         },
+#         {
+#             "watched": [
+#                 FANTASY_1,
+#                 ACTION_1,
+#                 INTRIGUE_1,
+#                 INTRIGUE_3,
+#             ]
+#         }
+#     ]
 
-# -----------------------------------------
-# ------------- WAVE 5 --------------------
-# -----------------------------------------
 
+def get_unique_watched(user_data):
+
+    unique = list(user_data["watched"])
+    friends = list(user_data["friends"])
+
+    for friend_dict in friends:
+        for movie in friend_dict["watched"]:
+            if movie in unique:
+                unique.remove(movie)
+
+    return unique
+
+
+def get_friends_unique_watched(user_data):
+
+    unique = []
+    friends = user_data["friends"]
+
+    for friend_dict in friends:
+        for movie in friend_dict["watched"]:
+            if (movie not in user_data["watched"]
+                    and movie not in unique):
+                unique.append(movie)
+
+    return unique
+    # -----------------------------------------
+    # ------------- WAVE 4 --------------------
+    # -----------------------------------------
+
+
+def get_available_recs(user_data):

```

Comment:
👍 

But we also have a function that we wrote that gets the movies that only the friends have watched which could simplify our logic here!